agent/http.go

```diff
@@ -427,39 +461,41 @@ func (s *HTTPHandlers) wrap(handler endpoint, methods []string) http.HandlerFunc
 					"error", err)
 			}
 
+			// If the error came from gRPC, unpack it to get the real message.
+			msg := err.Error()
+			if s, ok := status.FromError(err); ok {
+				msg = s.Message()
+			}
+
 			switch {
```

Comment:
I couldn't find unit tests for the error/status code conversion, but we should probably cover it with integration tests anyway.